### PR TITLE
Feature/icon functionality

### PR DESCRIPTION
**Pull request by artanmyrtolli**
# Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring

# How Has This Been Tested?
- [ ] open localHost
- [x] dev tools

# Checklist:
- [x] My code follows the Turing's guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] No console error messages

working on locking colors/icons

### CODE REVIEW

File: scripts.js [hunk 1]
```diff
@@ -2,18 +2,42 @@
...
-testFunction()
+newPalette.addEventListener('click', function(){
+	generateNewPalette()
 
-function testFunction() {
-	var currentPalette = new Palette()
-	for(var i = 0; i < 5; i++) {
+})
+
+
```
Comment (sted1994) [inline scripts.js:22]: Please get rid of empty space

File: scripts.js [hunk 2]
```diff
@@ -2,18 +2,42 @@
...
 var displayedHexCode = document.querySelectorAll('.hex-code')
+var paletteBox = document.querySelector('.palettes')
+var paletteId = document.querySelectorAll('.palettes__color')
 
+var currentPalette = new Palette();
+generateNewPalette()
 
 // Event Listeners:
+paletteBox.addEventListener('click', function(event){
+	console.log(event.target);
```
Comment (sted1994) [inline scripts.js:13]: Please remove consoles

**Review by sted1994 — APPROVED** [review]

File: scripts.js [hunk 3]
```diff
@@ -2,18 +2,42 @@
...
+	
+}
+
+function generateNewPalette() {
+	 currentPalette = new Palette();
+	for(var i = 0; i < currentColors.length; i++) {
 		currentColors[i].style.backgroundColor = currentPalette.colors[i].hexCode;
 		displayedHexCode[i].innerText = currentPalette.colors[i].hexCode;
+
+		currentColors[i].id = currentPalette.colors.indexOf(currentPalette.colors[i])
```
Comment (michaelputnam67) [inline scripts.js:40]: Could we add the ids to each individual element instead of to the parent?

**Review by michaelputnam67 — APPROVED** [review]
Remove empty space and build out lock functionality.

**Pull request merged**